packages/core/upload/server/src/services/upload.ts

```diff
@@ -553,9 +553,48 @@ export default ({ strapi }: { strapi: Core.Strapi }) => {
     });
   }
 
+  /**
+   * Uploads a file by path, buffer or stream programmaticaly
+   * @public
+   */
+  async function uploadFile(file: string | Buffer | fs.ReadStream, options: FileInfo) {
+    let filepath: string;
+
+    if (Buffer.isBuffer(file)) {
+      filepath = path.resolve(os.tmpdir(), crypto.randomBytes(10).toString('hex'));
+      await fse.writeFile(filepath, file);
+    } else if (file instanceof fs.ReadStream) {
+      filepath = path.resolve(os.tmpdir(), crypto.randomBytes(10).toString('hex'));
+      await fse.createWriteStream(filepath).write(file);
```

Comment:
You can't really "write" a readable stream to a writable stream though :thinking: 

Did you mean to pipe the readable into the writable?